source/common/io/io_uring_worker_impl.cc

```diff
@@ -400,15 +406,28 @@ void IoUringServerSocket::close() {
 
   IoUringSocketEntry::close();
 
-  if (read_req_ == nullptr && write_req_ == nullptr && cancel_req_ == nullptr) {
+  if (read_req_ == nullptr && write_req_ == nullptr && cancel_read_req_ == nullptr &&
+      cancel_write_req_ == nullptr) {
     ENVOY_LOG(trace, "ready to close, fd = {}", fd_);
     close_req_ = parent_.submitCloseRequest(*this);
     return;
   }
 
-  if (cancel_req_ == nullptr && read_req_ != nullptr) {
+  if (cancel_read_req_ == nullptr && read_req_ != nullptr) {
     ENVOY_LOG(trace, "cancel the read request, fd = {}", fd_);
-    cancel_req_ = parent_.submitCancelRequest(*this, read_req_);
+    cancel_read_req_ = parent_.submitCancelRequest(*this, read_req_);
+  }
+
+  if (write_timeout_timer_ == nullptr && cancel_write_req_ == nullptr && write_req_ != nullptr) {
```

Comment:
also, write_timeout_timer_ must be nullptr here, right?

If yes, we probably should use assertion here.